src/main/java/com/github/manolo8/darkbot/core/utils/factory/EntityFactory.java

```diff
@@ -62,18 +93,34 @@ public static EntityFactory find(long address, int id) {
                     getZoneKey(address) : assetId).matches()) return type;
         }
 
-        return isShip(address, id) ? SHIP : UNKNOWN;
+        return isPet(address) ? PET : isShip(address, id) ? SHIP : UNKNOWN;
+    }
+
+    private static EntityFactory defineZoneType(long address) {
+        String key = getZoneKey(address);
+        return (key.equals("NOA")) ? BARRIER : key.equals("DMG") ? MIST_ZONE : UNKNOWN;
+    }
+
+    private static EntityFactory defineShipType(long address) {
+        int isNpc = API.readMemoryInt(address + 112);
+
+        return isNpc == 1 ? NPC : isNpc == 0 && address != HeroManager.instance.address &&
+                address != HeroManager.instance.pet.address ? SHIP : UNKNOWN; //fix
     }
```

Comment:
I thought these would make more sense as `isZone`, or `isMistZone` functions, that just return a boolean. So they're `Predicate<Long>` instead of `Function<Long, EntityFactory>`. It's a bit weird that one entity factory enum entry can tell you to go to another one, feels like it makes more sense for it to just return the true/false as if that one matches properly